### PR TITLE
[serve] Migrate v1 api tests

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Migration for non-serve tests and release tests will be in follow up PRs.

Also, I didn't migrate the tests in `test_deployment_graph_build` because it isn't super clear to me how to - will also address in a follow up PR.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_deploy_2.py [hunk 1]
```diff
@@ -13,200 +12,10 @@
...
-    @serve.deployment(name="hi", num_replicas=2)
-    def d1(*args):
-        pass
-
-    d1._deploy()
-
-    assert _private_api.list_deployments() == {"hi": d1}
-
-
-def test_deploy_change_route_prefix(serve_instance):
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy_2.py:172]: Migrated to `test_deploy_app.py::test_change_route_prefix`

File: python/ray/serve/tests/test_handle.py [hunk 2]
```diff
@@ -208,57 +205,63 @@ def test_repeated_get_handle_cached(serve_instance):
...
+        assert kwargs["kwarg1"] == 1
+        assert kwargs["kwarg2"] == "2"
+
+    handle = serve.run(f.bind()).options(use_new_handle_api=True)
+    handle.remote("hi", kwarg1=1, kwarg2="2").result()
+
+
 @pytest.mark.asyncio
-@pytest.mark.parametrize("sync", [True, False])
-async def test_args_kwargs(serve_instance, sync):
```
Comment (zcin) [inline python/ray/serve/tests/test_handle.py:222]: Split into `test_args_kwargs_sync` and `test_args_kwargs_async`

File: python/ray/serve/tests/test_handle.py [hunk 3]
```diff
@@ -208,57 +205,63 @@ def test_repeated_get_handle_cached(serve_instance):
...
 
-    ray.get(obj_ref)
+    exception_string = str(excinfo.value)
+    assert "'does_not_exist'" in exception_string
+    assert "Available methods: ['exists']" in exception_string
 
 
 @pytest.mark.asyncio
-@pytest.mark.parametrize("sync", [True, False])
-async def test_nonexistent_method(serve_instance, sync):
```
Comment (zcin) [inline python/ray/serve/tests/test_handle.py:246]: Split into `test_nonexistent_method_sync` and `test_nonexistent_method_async`

File: python/ray/serve/tests/test_deploy.py [hunk 4]
```diff
@@ -81,105 +75,6 @@ def ping(self, *args):
     assert ray.get(class_handle.ping.remote()) == "pong"
 
 
-@pytest.mark.parametrize("use_handle", [True, False])
-def test_deploy_no_version(serve_instance, use_handle):
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy.py:85]: Version only works with v1 api

File: python/ray/serve/tests/test_deploy.py [hunk 5]
```diff
@@ -81,105 +75,6 @@ def ping(self, *args):
...
-    assert pid4 != pid3
-
-    v2.options(version="1")._deploy()
-    val5, pid5 = call()
-    assert val5 == "2"
-    assert pid5 == pid4
-
-
-@pytest.mark.parametrize("use_handle", [True, False])
-def test_config_change(serve_instance, use_handle):
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy.py:132]: User config lightweight change covered by tests in test_deploy_app.py

File: python/ray/serve/tests/test_runtime_env.py [hunk 6]
```diff
@@ -98,62 +98,6 @@ def __call__(self, *args):
     run_string_as_driver(driver2)
 
 
-# NOTE: This test uses deployment.deploy() instead of serve.run() to preserve
-# the cached runtime_env that's returned by serve.get_deployment().
-@pytest.mark.skipif(sys.platform == "win32", reason="Fail to create temp dir.")
-def test_working_dir_scale_up_in_new_driver(ray_start, tmp_dir):
```
Comment (zcin) [inline python/ray/serve/tests/test_runtime_env.py:104]: Synced with Shreyas, this test is no longer relevant.

File: python/ray/serve/tests/test_http_routes.py [hunk 7]
```diff
@@ -48,52 +46,6 @@ def test_routes_healthz(serve_instance):
     assert resp.content == b"success"
 
 
-def test_routes_endpoint_legacy(serve_instance):
```
Comment (zcin) [inline python/ray/serve/tests/test_http_routes.py:51]: There is already a v2 api version of this test below, removing the legacy test

File: python/ray/serve/tests/test_deploy_2.py [hunk 8]
```diff
@@ -13,200 +12,10 @@
 from ray import serve
 from ray._private.pydantic_compat import ValidationError
 from ray._private.test_utils import SignalActor, wait_for_condition
-from ray.serve._private import api as _private_api
 from ray.serve._private.common import ApplicationStatus
 from ray.serve.drivers import DAGDriver
 
 
-class TestGetDeployment:
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy_2.py:21]: Removing all `get_deployment` and `list_deployments` tests

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@zcin serve test failures in premerge build

**Comment by zcin** [comment]
@edoakes Tests are passing (currently failing tests are failing on master):
Screen Shot 2023-10-17 at 7 59 50 AM

**Pull request merged**